third_party/spdk_defs/nvme_defs.h

```diff
@@ -239,6 +239,128 @@ struct SglDescriptor {
 } ABSL_ATTRIBUTE_PACKED;
 static_assert(sizeof(SglDescriptor) == 16);
 
+// NVMe Base Specification Figure 245
+// https://nvmexpress.org/wp-content/uploads/NVM-Express-1_4-2019.06.10-Ratified.pdf
+struct IdentifyNamespace {
+  uint64_t nsze;  // namespace size
+  uint64_t ncap;  // namespace capacity
+  uint64_t nuse;  // namespace utilization
+
+  struct {
+    uint8_t thin_prov : 1;             // thin provisioning
+    uint8_t ns_atomic_write_unit : 1;  // NAWUN, NAWUPF, and NACWU are defined
+                                       // for this namespace
+    uint8_t
+        dealloc_or_unwritten_error : 1;  // Supports Deallocated or Unwritten
```

Comment:
nit: I'd prefer to shorten either comments or variable name so that field type & name are on the same line